### ISSUE TITLE
[solved] Q not PSD error with Gurobi 0.9?

### ISSUE THREAD

**Issue by NikitaBelyakAalto** [JuliaOpt/Gurobi.jl#285]
The package Gurobi.jl seems to be not updated in accordance with the current release of the software Gurobi 9.0. As it has been announced on the official [website](https://www.gurobi.com/products/gurobi-optimizer/whats-new-current-release/) Gurobi 9.0 is already capable of solving non-convex quadratic problems, however, when trying to use the latest version of the software to solve JuMP MIQCQP I receive the following message 
![image](https://user-images.githubusercontent.com/53814737/73745114-ef8f5c00-475a-11ea-941c-093d646ecf35.png)
Could you please have a look at this issue? 
Please find attached the .pdf with the formulation of the model I used and .jl with the JuMP formulation of it in the enclosed archive [Non-convex MIQCQP example.zip](https://github.com/JuliaOpt/Gurobi.jl/files/4154178/Non-convex.MIQCQP.example.zip)

![Screenshot from the gurobi website ](https://user-images.githubusercontent.com/53814737/73756629-8b779280-4770-11ea-9fd5-fa6c15fce9fe.png)

**Comment by odow** [comment]
You need to set the `NonConvex` parameter to 2. See: https://www.gurobi.com/documentation/9.0/refman/nonconvex.html

```Julia
model = Model(with_optimizer(Gurobi.Optimizer, NonConvex = 2))
```

Given the number of complaints about this, it's a little annoying that Gurobi doesn't highlight this more in their documentation.

**Comment by odow** [comment]
I've changed the title because there will undoubtedly be more people that come looking for this question.

**Comment by NikitaBelyakAalto** [comment]
Thank you very much for your help. I just tried that and I got the following 
![image](https://user-images.githubusercontent.com/53814737/73761746-ee205c80-4777-11ea-9b3b-b02b6051854b.png)
Do you know by any chance how to fix this?

**Comment by odow** [comment]
You need to ensure you have the latest tagged version of Gurobi.jl (0.7.5).

Run
```Julia
import Pkg
Pkg.update()
Pkg.status()
```

**Comment by NikitaBelyakAalto** [comment]
received this [2e9cd046] Gurobi v0.7.4
Tried even Pkg.update("Gurobi") but the version remains the same

**Comment by odow** [comment]
You likely have another package holding you back.

What's the output of `Pkg.status()`?

In general, you should use Julia environments to avoid this: https://julialang.github.io/Pkg.jl/stable/environments/

**Comment by NikitaBelyakAalto** [comment]
julia> Pkg.status()
    Status `~/.julia/environments/v1.2/Project.toml`
  [07493b3f] Alpine v0.1.3
  [7c4d4715] AmplNLWriter v0.5.0
  [c52e3926] Atom v0.11.3
  [336ed68f] CSV v0.5.23
  [9961bab8] Cbc v0.6.6
  [a93c6f00] DataFrames v0.20.0
  [31c24e10] Distributions v0.22.4
  [f6369f11] ForwardDiff v0.10.9
  [2e9cd046] Gurobi v0.7.4
  [7073ff75] IJulia v1.21.0
  [b6b21f68] Ipopt v0.6.1
  [4076af6c] JuMP v0.20.1
  [e5e0dc1b] Juno v0.7.2
  [f4570300] MathOptFormat v0.4.0
  [0db19996] NBInclude v2.1.0
  [76087f3c] NLopt v0.5.1
  [91a5bcdd] Plots v0.28.4
  [d330b81b] PyPlot v2.8.2
  [fdbf4ff8] XLSX v0.5.8
  [9a3f8284] Random 
  [2f01184e] SparseArrays

**Comment by NikitaBelyakAalto** [comment]
Thank you,  I will use a separate project further

**Comment by odow** [comment]
You can also try `] add Gurobi@0.7.5` and see what the error is. 

In general, it isn't recommended to have this many packages in your global environment for precisely this reason. For every project you do, make a new environment with the minimal subset of packages you need.

For example, Alpine and AmplNLWriter are installed at versions which don't support JuMP 0.20. It gets pretty confusing if lots of packages are all fighting for version requirements.

**Comment by NikitaBelyakAalto** [comment]
Thank you very much for your help. I just solved it

**Comment by odow** [comment]
In good news, the next version of Gurobi will have a more informative error message.

**Comment by odow** [comment]
Closing. Fixed by new error message in Gurobi 9.0.2.
